src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -383,6 +397,103 @@ public String updateAccount(UserProvenances userProvenance, String provenanceUse
+     *
+     * @param pageable The pageable object.
+     * @param email The email to query by.
+     * @param userProvenanceId The user provenance id to query by.
+     * @param userProvenances A list of user provenances to query by.
+     * @param roles A list of roles to query by.
+     * @param userId The user id to query by.
+     * @return A page with a list of piUsers.
+     */
+    public Page<PiUser> findAllAccountsExceptThirdParty(Pageable pageable, String email, String userProvenanceId,
+                                                        List<UserProvenances> userProvenances, List<Roles> roles,
+                                                        String userId) {
+
+        // If a user ID is supplied then only find by that
+        if (!userId.isBlank()) {
+            return userRepository.findByUserIdPageable(userId, pageable);
+        }
+
+        //  If a list of user provenances are supplied then use that else use all non third party user provenances
+        List<UserProvenances> userProvenancesToQuery = ALL_NON_THIRD_PARTY_PROVENANCES;
+        if (!userProvenances.isEmpty()) {
+            userProvenancesToQuery = userProvenances;
+        }
+
+        // If a list of roles are supplied then use that else use all non third party roles
+        List<Roles> rolesToQuery = ALL_NON_THIRD_PARTY_ROLES;
+        if (!roles.isEmpty()) {
+            rolesToQuery = roles;
+        }
+
+        // If user provenance id is supplied then find by an exact match
+        String userProvenanceIdToQuery = "%%";
+        if (!userProvenanceId.isBlank()) {
+            userProvenanceIdToQuery = userProvenanceId;
+        }
+
+        return userRepository.findAllByEmailLikeAndUserProvenanceInAndRolesInAndProvenanceUserIdLike(
+            "%" + email + "%",
+            userProvenancesToQuery,
+            rolesToQuery,
+            userProvenanceIdToQuery,
+            pageable
+        );
+    }
+
+    /**
+     * Get a user by a user id.
+     * @param userId The users id to search by.
+     * @return A pi user object.
+     */
+    public PiUser getUserById(UUID userId) {
+        return userRepository.findByUserId(userId).orElseThrow(() -> new NotFoundException(String.format(
+            "User with supplied user id: %s could not be found", userId)));
+    }
+
+    /**
+     * Process a manual user deletion request.
+     *
+     * @param userId The ID of the user to delete.
+     */
+    public void processManualAccountDeletion(UUID userId) {
+        PiUser userToDelete = userRepository.findByUserId(userId).orElseThrow(() -> new NotFoundException(String.format(
+            "User with supplied user id: %s could not be found", userId)));
+
+        deleteAccount(userToDelete.getEmail(), UserProvenances.PI_AAD.equals(userToDelete.getUserProvenance()));
+    }
+
+    /**
+     * Process updating a role for an account.
+     *
+     * @param userId The ID of the user to update.
+     */
```

Comment:
Just something really minor. Need the second param and the return in the header of this method

Reply:
Done :)